### ISSUE TITLE
The Weighted_GAP  function

### ISSUE THREAD

**Issue by Hai-Jun-Yan** [luoxiaoliu/PFENet2Plus#2]
Hello, I hope this message finds you well.

I noticed a function called Weighted_GAP in both PFENet.py and PFENet2Plus.py. I'm curious about the specific purpose of this function and how the parameter 0.0005 within it is determined. Could you please shed some light on its specific role and the rationale behind setting the parameter to 0.0005?

Thank you for your time and insights!

**Comment by xiaowillow** [comment]
Hello, thank you for your interest. The purpose of the Weighted_GAP function is to perform global average pooling weighted by a mask. The value 0.005 is added to prevent the calculated area from being zero when the mask contains only zeros, as the area will be used as the denominator in the next step and cannot be zero.

**Comment by Hai-Jun-Yan** [comment]
Thank you very much for your reply. I've noticed that many research papers on few-shot semantic segmentation utilize this function in their code. However, I haven't been able to find its origin. I'm very curious about its source, as I believe its difference from the "Masked Average Pooling" proposed in Sg-One lies solely in the pooling algorithm. Do you happen to know its origin?I apologize if I'm asking a basic question，and any information you can share would be greatly appreciated.

**Comment by Hai-Jun-Yan** [comment]
I'm working on a project involving 1-shot and 5-shot models, and I was wondering if you have any insights on whether the  hyperparameters like the learning rate should be different for each. Your expertise is highly valued!
